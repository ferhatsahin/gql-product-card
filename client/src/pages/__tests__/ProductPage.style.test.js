import React from "react";
import { render, cleanup } from "../../test.utils";

import {
  PageContainer,
  InfoTab,
  Container,
  HeartIcon
} from "../ProductPage.style";

afterEach(cleanup);

test("PageContainer style test", () => {
  const { container } = render(<PageContainer />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule("justify-content", "center");
  expect(container.firstChild).toHaveStyleRule("flex-direction", "column");
});

test("InfoTab style test", () => {
  const { container } = render(<InfoTab />);
  expect(container.firstChild).toMatchSnapshot();
});

test("Container style test", () => {
  const { container } = render(<Container />);
  expect(container.firstChild).toMatchSnapshot();
});

test("HeartIcon default style test", () => {
  const { container } = render(<HeartIcon />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule("color", "gray");
});
