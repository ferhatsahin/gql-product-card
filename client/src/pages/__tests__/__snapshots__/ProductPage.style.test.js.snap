// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container style test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media only screen and (max-width:640px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div
  class="c0"
/>
`;

exports[`HeartIcon default style test 1`] = `
.c0 {
  width: 30px;
  color: gray;
  padding-right: 10px;
}

<svg
  class="c0"
>
  heart.svg
</svg>
`;

exports[`InfoTab style test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
  padding: 20px;
  font-size: 25px;
  margin-right: 25px;
  margin-top: 10px;
}

<div
  class="c0"
/>
`;

exports[`PageContainer style test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 7%;
  background: slategray;
}

<div
  class="c0"
/>
`;
